README.md

```diff
@@ -1,2 +1,331 @@
+
+## 2.2 Signed Messages
+
+Signed Messages are cryptographically signed messages from an account that are **tamper-proof** and **self-authenticating**. 
+
+A Signed Message has a **message** object, which contains the data that will be signed. The message is then serialized, and then hashed and signed with the custody address. The hash, signature and signer public key are used to construct the **envelope**, which proves that the message is authentic and has not been modified. Any observer can verify the message by checking that: 
+
+1. The account number is owned by an address whose public key matches the public key in the envelope
+2. The serialized hash of message matches the hash in the envelope
+3. The signature in the envelope is verified by the signing scheme with the hash and public key.   
+
+Messages must contain an account number, to look up the `signerPubKey` on-chain, and a `timestamp`, to order messages. Timestamps are client provided, unverified and should be considered best-effort, since they are vunlerable to [clock skew](https://en.wikipedia.org/wiki/Clock_skew) and [clock drift](https://en.wikipedia.org/wiki/Clock_drift). Users who want to ensure perfect ordering can use [hybrid clocks](https://martinfowler.com/articles/patterns-of-distributed-systems/hybrid-clock.html) to generate timestamps.
+
+```ts
+type SignedMessage = {
+  message: {
+    body: any; 
+    account: number;
+    timestamp: number;
+  };
+  envelope: {
+    hash: string;
+    hashType: 'BLAKE2b';
+    signature: string;
+    signatureType: 'ed25519' | 'ecdsa-secp256k1';
+    signerPubKey: string;
+  }
+};
+```
+
+Object serialization must be done according to [RFC-8785](https://datatracker.ietf.org/doc/html/rfc8785), hashing must be performed with [BLAKE2b](https://www.rfc-editor.org/rfc/rfc7693.txt) and signing must be performed with an Ed25519 signature scheme, or in rare cases an ECDSA secp256k1 scheme. 
+
+
+## 2.3 Applications
+An application is a program that people use to interact with the Farcaster network. It can be standalone client that can be downloaded and run locally, or hosted software that runs on a cloud server and can be accessed through a specific client. Users are free to choose the type of application that best suits their needs. 
+
+A simple application might consist of a standalone desktop or mobile client that talks directly to a Farcaster Hub. It can publish new messages and view messages published by other accounts. Each client will need to be instantiated with a [valid signing key](#42-signer). 
+
+A more sophisticated application might introduce a proxy backend server between the Hub and the client. Advanced features like search can be implemented on this server. It might also operate as a hosted service, where it custodies keys for the user and performs all the signing on the backend.
```

Comment:
Random suggestion: maybe a better example is "bring your own algorithm" clients? Or different moderation strategies?